File: src/main/kotlin/models/Task.kt
package com.todoapp.models

import kotlinx.serialization.Serializable
import java.time.LocalDateTime

@Serializable
data class Task(
    val id: Long,
    val title: String,
    val description: String?,
    val isCompleted: Boolean = false,
    val parentId: Long? = null,
    val position: Int = 0,
    val children: List<Task> = emptyList(),
    val createdAt: String,
    val updatedAt: String
)

File: src/types/Task.ts
export interface Task {
  id: number;
  title: string;
  description?: string;
  isCompleted: boolean;
  parentId?: number;
  position: number;
  children: Task[];
  createdAt: string;
  updatedAt: string;
}

export interface CreateTaskRequest {
  title: string;
  description?: string;
  parentId?: number;
  position?: number;
}

export interface UpdateTaskRequest {
  title?: string;
  description?: string;
  isCompleted?: boolean;
  position?: number;
}

export interface TaskResponse {
  success: boolean;
  message: string;
  data?: Task;
}

export interface TaskListResponse {
  success: boolean;
  message: string;
  data: Task[];
}
